dev-packages/application-package/src/application-package.ts

```diff
@@ -130,11 +130,11 @@ export class ApplicationPackage {
 
     async resolveExtensionPackage(extension: string): Promise<ExtensionPackage | undefined> {
         const raw = await RawExtensionPackage.view(this.registry, extension);
-        return raw ? this.newExtensionPackage(raw) : undefined;
+        return raw ? this.newExtensionPackage(raw, extension) : undefined;
     }
 
-    protected newExtensionPackage(raw: PublishedNodePackage): ExtensionPackage {
-        return new ExtensionPackage(raw, this.registry);
+    protected newExtensionPackage(raw: PublishedNodePackage, name: string): ExtensionPackage {
+        return new ExtensionPackage(raw, this.registry, name);
```

Comment:
Rename `name` to `alias`? It can be confusing why we need to pass this string around instead of doing `raw.name`.

We could update the other locations in the same way.